Source/Core/Core/IOS/USB/Emulated/Infinity.cpp

```diff
@@ -0,0 +1,958 @@
+      // II is the order the figure was added (00 for first, 01 for second and 03 for third).
+      // BB is the block number. 00 is block 1, 01 is block 4, 02 is block 8, and 03 is block 12).
+      // UU is either 00 or 01 -- not exactly sure what it indicates.
+      infinity_base.QueryBlock(buf[4], buf[5], response_data.data(), sequence);
+      break;
+    }
+
+    case 0xA3:
+    {
+      // A3 - Write Figure Data:
+      // This writes a block of 16 bytes to a figure on the infinity base.
+      // Attached data: II BB UU <16 bytes>
+      // II is the order the figure was added (00 for first, 01 for second and 03 for third).
+      // BB is the block number. 00 is block 1, 01 is block 4, 02 is block 8, and 03 is block 12).
+      // UU is either 00 or 01 -- not exactly sure what it indicates.
+      // <16 bytes> is the raw (encrypted) 16 bytes to be sent to the figure.
+      infinity_base.WriteBlock(buf[4], buf[5], &buf[7], response_data.data(), sequence);
+      break;
+    }
+
+    case 0xB4:
+    {
+      // B4 - Get Tag ID:
+      // This gets the tag's unique ID for a figure on the infinity base.
+      // The first byte is the order the figure was added (00 for first, 01 for second and 03 for
+      // third). The infinity base will respond with 7 bytes (the tag ID).
+      infinity_base.GetFigureIdentifier(buf[4], sequence, response_data.data());
+      break;
+    }
+
+    case 0xB5:
+    {
+      // Get status?
+      infinity_base.GetBlankResponse(sequence, response_data.data());
+      break;
+    }
+
+    default:
+      break;
+    }
+    memcpy(data.data(), buf, 32);
+    ScheduleTransfer(std::move(cmd), data, 32, 500);
+    if (m_response_list.empty())
+    {
+      m_queries.push(response_data);
+    }
+    else
+    {
+      ScheduleTransfer(std::move(m_response_list.front()), response_data, 32, 1000);
+      m_response_list.pop();
+    }
+  }
+  return 0;
+}
+int InfinityUSB::SubmitTransfer(std::unique_ptr<IsoMessage> cmd)
+{
+  DEBUG_LOG_FMT(IOS_USB,
+                "[{:04x}:{:04x} {}] Isochronous: length={:04x} endpoint={:02x} num_packets={:02x}",
+                m_vid, m_pid, m_active_interface, cmd->length, cmd->endpoint, cmd->num_packets);
+  return 0;
+}
+void InfinityUSB::ScheduleTransfer(std::unique_ptr<TransferCommand> command,
+                                   const std::array<u8, 32>& data, s32 expected_count,
+                                   u64 expected_time_us)
+{
+  command->FillBuffer(data.data(), expected_count);
+  command->ScheduleTransferCompletion(expected_count, expected_time_us);
+}
+
+bool InfinityBase::HasFigureBeenAddedRemoved() const
+{
+  return !m_figure_added_removed_response.empty();
+}
+
+std::array<u8, 32> InfinityBase::GetAddedRemovedResponse()
+{
+  std::array<u8, 32> response = m_figure_added_removed_response.front();
+  m_figure_added_removed_response.pop();
+  return response;
+}
+
+u8 InfinityBase::GenerateChecksum(u8* data, int num_of_bytes) const
```

Comment:
```suggestion
u8 InfinityBase::GenerateChecksum(const u8* data, int num_of_bytes) const
```
(or even `std::span<const u8>`)